### PR TITLE
Add ClusterOperator

### PR DESCRIPTION
**Pull request by wongma7**
Missed this: https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md

### CODE REVIEW

**Comment by wongma7** [comment]
@openshift/storage  PTAL at last commit, specifically https://github.com/openshift/cluster-storage-operator/pull/10/commits/ae4ef0bb5c14dc158d28539ab99da9eab54de91d#diff-97e82b0d41902abcb7a788ea37573993

we will report either failing or available. keeping it simple...

**Comment by wongma7** [comment]
{"level":"info","ts":1549648529.729961,"logger":"controller_clusterstorage","msg":"Reconciling ConfigMap","Request.Namespace":"kube-system","Request.Name":"cluster-config-v1"}
E0208 17:57:28.087998       1 streamwatcher.go:109] Unable to decode an event from the watch stream: http2: server sent GOAWAY and closed the connection; LastStreamID=277, ErrCode=NO_ERROR, debug=""
E0208 17:57:28.107809       1 streamwatcher.go:109] Unable to decode an event from the watch stream: http2: server sent GOAWAY and closed the connection; LastStreamID=277, ErrCode=NO_ERROR, debug=""
E0208 17:57:28.108436       1 streamwatcher.go:109] Unable to decode an event from the watch stream: http2: server sent GOAWAY and closed the connection; LastStreamID=277, ErrCode=NO_ERROR, debug=""

File: manifests/03-cluster-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: config.openshift.io/v1
+kind: ClusterOperator
+metadata:
+  name: cluster-storage-operator
+  spec: {}
```
Comment (gnufied) [inline manifests/03-cluster-operator.yaml]: In the linked document, this appears to be a namespaced resource?

Comment (wongma7) [reply manifests/03-cluster-operator.yaml]: they aren't, seems the doc is outdated.  Also, there is a `versions` array, not a single `version` now.

Comment (wongma7) [reply manifests/03-cluster-operator.yaml]: hmm this is suspect: " Also note that the ClusterOperator resource in /manifests is only a communication mechanism, to tell the ClusterVersionOperator, which ClusterOperator resource to wait for. The ClusterVersionOperator does not create the ClusterOperator resource, this and updating it is the responsibility of the respective operator."

**Comment by wongma7** [comment]
Might be working now, will test a few more times

        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "ClusterOperator",
            "metadata": {
                "creationTimestamp": "2019-02-08T22:36:25Z",
                "generation": 1,
                "name": "cluster-storage-operator",
                "resourceVersion": "10214",
                "selfLink": "/apis/config.openshift.io/v1/clusteroperators/cluster-storage-operator",
                "uid": "f77af8c5-2bf1-11e9-bf02-12814fa94314"
            },
            "spec": {},
            "status": {
                "conditions": [
                    {
                        "lastTransitionTime": "2019-02-08T22:36:25Z",
                        "status": "False",
                        "type": "Progressing"
                    },
                    {
                        "lastTransitionTime": "2019-02-08T22:36:25Z",
                        "status": "True",
                        "type": "Available"
                    },
                    {
                        "lastTransitionTime": "2019-02-08T22:36:25Z",
                        "status": "False",
                        "type": "Failing"
                    }
                ],
                "extension": null,
                "relatedObjects": null,
                "versions": [
                    {
                        "name": "operator",
                        "version": "0.0.1"
                    }
                ]
            }
        },

**Comment by wongma7** [comment]
PTAL, green:

        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "ClusterOperator",
            "metadata": {
                "creationTimestamp": "2019-02-11T20:33:50Z",
                "generation": 1,
                "name": "cluster-storage-operator",
                "resourceVersion": "9744",
                "selfLink": "/apis/config.openshift.io/v1/clusteroperators/cluster-storage-operator",
                "uid": "57226f8d-2e3c-11e9-96c8-0e01cb555bec"
            },
            "spec": {},
            "status": {
                "conditions": [
                    {
                        "lastTransitionTime": "2019-02-11T20:33:50Z",
                        "status": "False",
                        "type": "Progressing"
                    },
                    {
                        "lastTransitionTime": "2019-02-11T20:33:50Z",
                        "status": "True",
                        "type": "Available"
                    },
                    {
                        "lastTransitionTime": "2019-02-11T20:33:50Z",
                        "status": "False",
                        "type": "Failing"
                    }
                ],
                "extension": null,
                "relatedObjects": null,
                "versions": [
                    {
                        "name": "operator",
                        "version": "0.0.1"
                    }
                ]
            }
        },

File: pkg/controller/clusterstorage/clusterstorage_controller.go [hunk 2]
```diff
@@ -123,9 +125,31 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	clusterOperatorInstance := &configv1.ClusterOperator{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-storage-operator",
+			Namespace: corev1.NamespaceAll,
```
Comment (gnufied) [inline pkg/controller/clusterstorage/clusterstorage_controller.go]: Do we need to set this all?

**Comment by wongma7** [comment]
bunch of "
    watch closed before UntilWithoutRetry timeout" errors https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-storage-operator/10/pull-ci-openshift-cluster-storage-operator-master-e2e-aws/37
/retest

**Pull request merged**